paddle/fluid/operators/mkldnn/layer_norm_mkldnn_op.cc

```diff
@@ -37,49 +37,34 @@ class LayerNormOneDNNHandler
             engine, cpu_place) {
     const auto fwd_prop_kind = is_test ? dnnl::prop_kind::forward_inference
                                        : dnnl::prop_kind::forward_training;
+
     this->AcquireForwardPrimitiveDescriptor(
-        fwd_prop_kind, x->mem_desc(), epsilon, flags);
+        fwd_prop_kind, x->mem_desc(), x->mem_desc(), epsilon, flags);
```

Comment:
Because onednn v3.x layernorm primitive desc creation api has been changed to use both src md and dst md, to allow more optimization opportunities.

Pls see here: https://github.com/oneapi-src/oneDNN/blob/master/include/oneapi/dnnl/dnnl.hpp#L7638